language/stdlib/modules/multi_account.move

```diff
@@ -0,0 +1,104 @@
+// Errors:
+// 0 -> ADDRESS_NOT_ALLOWED
+// 1 -> CURRENCY_NOT_HELD
+address 0x0:
+module MultiCurrencyHolding {
+    use 0x0::LibraAccount;
+    use 0x0::Transaction;
+    use 0x0::Libra;
+    use 0x0::Vector;
+
+    // A resource that specifies the addresses that are allowed to be added
+    // to this multi-currency holding.
+    resource struct ChildInfo {
+        allowed_addresses: vector<address>,
+    }
+
+    // Withdrawal capabilities for the added accounts, partitioned by their
+    // currency type.
+    resource struct AccountInfo<Token> {
+        withdrawal_caps: vector<LibraAccount::WithdrawalCapability>,
+    }
+
+    // Allow the account at `addr` to delegate its
+    // withdrawal_capability to us.
+    public fun allow_account_address(addr: address)
+    acquires ChildInfo {
+        let sender = Transaction::sender();
+        if (!exists<ChildInfo>(sender)) move_to_sender(ChildInfo{
```

Comment:
Might be cleaner to have a separate proc for this instead of using lazy initialization?